core/CronArchive.php

```diff
@@ -1010,13 +1088,24 @@ private function archiveReportsFor($idSite, $period, $date, $archiveSegments, Ti
 
             $self = $this;
             $request = new Request($url);
-            $request->before(function () use ($self, $url, $idSite, $period, $date) {
+            $request->before(function () use ($self, $url, $idSite, $period, $date, $segment, $request) {
                 if ($self->isAlreadyArchivingUrl($url, $idSite, $period, $date)) {
-                     return Request::ABORT;
+                    return Request::ABORT;
+                }
+
+                list($isThereArchive, $newDate) = $this->isThereAValidArchiveForPeriod($idSite, $period, $date, $segment);
+                if ($isThereArchive) {
+                    $this->logArchiveWebsiteSkippedValidArchiveExists($idSite, $period, $date);
+                    return Request::ABORT;
                 }
+
+                $url = $request->getUrl();
```

Comment:
@diosmosis any chance to add a test for this? could maybe move this somehow into the request class and then call like 

```php
$request->changeDate($newDate)
```

and this changeDate method could be then public and testable? with regexp I always get nervous so be good to have a basic test for it. Otherwise there doesn't seem anything missing anymore?

Reply:
Moved and added a test